### PR TITLE
[RIC] 1.4.1 Related Product Cards

### PR DESCRIPTION
**Pull request by bishalkg**
[RIC] 1.4.1 Related Product Cards on Trello. With implementation tasks 1.4.1.1, 1.4.1.2, 1.4.1.3. 
The files within 'Related Products' and 'Related Outfits' are virtually the same with slight different functionality. I separated these out here, but they will be consolidated for code-reuse in a later refactoring.

### CODE REVIEW

File: client/src/components/RelatedItemsAndOutfit/RelatedItemsAndOutfit.jsx [hunk 1]
```diff
@@ -1,18 +1,64 @@
 import React from 'react';
+import server from './Axios';
+import RelatedProducts from './RelatedProducts/RelatedProducts';
+import UserOutfit from './UserOutfits/UserOutfit';
 
 class RelatedItemsAndOutfit extends React.Component {
   constructor(props) {
     super(props);
-
+    this.server = server;
```
Comment (ya-liu) [inline client/src/components/RelatedItemsAndOutfit/RelatedItemsAndOutfit.jsx]: Do you absolutely need to attach server to this component? The server is an imported axios instance, does it work without 'this' on line 24 minus this declaration on line 9?

**Review by ya-liu — COMMENTED** [review]
Just had a question about this.server, the rest look really good. I like the touch with 'Loading...', very detail-oriented!

File: client/src/components/RelatedItemsAndOutfit/RelatedItemsAndOutfit.jsx (hunk 1 above)
Comment (bishalkg) [reply client/src/components/RelatedItemsAndOutfit/RelatedItemsAndOutfit.jsx]: Yeah so I had it like that originally, but if I don't attach it to the class itself, esLint AirBnB gets angry and says 'must use 'this' in methods of this class', so I had to attach it to the constructor

**Comment by bishalkg** [comment]
> Just had a question about this.server, the rest look really good. I like the touch with 'Loading...', very detail-oriented!

OOh important to note, I implemented the loading conditional because I ran into an issue:
I had to do this so that it wouldn't render twice! (Once for the initial Render React does, and once for render with onComponentDidMount). If I didn't have this it would pass in empty arrays as props (from my state) on the initial Render, and then pass in the fetched arrays (once filled in state) on the ComponentDidMount Render, but I only want to pass the props after the fetch (bc I get potential errors downstream if I pass an empty array)

File: client/src/components/RelatedItemsAndOutfit/RelatedItemsAndOutfit.jsx (hunk 1 above)
Comment (bishalkg) [reply client/src/components/RelatedItemsAndOutfit/RelatedItemsAndOutfit.jsx]: I couldn't replicate it, very curious! But the error was  "warning  Expected 'this' to be used by class method". I changed it to not use 'this'.

**Review by dho1994 — APPROVED** [review]

**Review by ya-liu — APPROVED** [review]

**Pull request merged**